### PR TITLE
Added support of nine digit nanoseconds format.

### PR DESCRIPTION
**Pull request by ndallmeyer**
Added support for up to nine digit nanoseconds format. The nanoseconds are truncated to 7 decimal places. See https://github.com/kubernetes-client/csharp/issues/1350

### CODE REVIEW

**Comment by tg123** [comment]
could you please merge my testcase

**Comment by ndallmeyer** [comment]
Thanks, based on your tests I've implemented tests that cover all edge cases.

File: src/KubernetesClient.Models/KubernetesJson.cs [hunk 1]
```diff
@@ -26,18 +26,49 @@ public override void Write(Utf8JsonWriter writer, TimeSpan value, JsonSerializer
 
         private sealed class KubernetesDateTimeOffsetConverter : JsonConverter<DateTimeOffset>
         {
-            private const string SerializeFormat = "yyyy'-'MM'-'dd'T'HH':'mm':'ss.ffffffK";
-            private const string Iso8601Format = "yyyy'-'MM'-'dd'T'HH':'mm':'ssK";
+            private const string RFC3339NanoFormat = "yyyy-MM-dd'T'HH':'mm':'ss.fffffffK";
+
+            private static string FormatDateTimeOffsetToSevenDigitsNanoseconds(string dateTime)
+            {
+                var isUTC = dateTime.EndsWith("Z");
```
Comment (tg123) [inline src/KubernetesClient.Models/KubernetesJson.cs:33]: too complex
why not just trim extra input?

Comment (ndallmeyer) [reply src/KubernetesClient.Models/KubernetesJson.cs:33]: As already pointed out by @anekdoti in https://github.com/kubernetes-client/csharp/issues/1350 the lastTransitionTimestamp should be deserializable as a Golang Time struct that follows RFC3339 where an arbitrary number of nansecond digits from [0 to 9 is valid](https://cs.opensource.google/go/go/+/refs/tags/go1.20.6:src/time/format.go;l=1494). Therefore trimming alone would not help here.
This method allows an arbitrary number of nanosecond digits from 0 to 9.

**Comment by tg123** [comment]
check my impl in #1351

**Comment by anekdoti** [comment]
Hello @tg123,

**Comment by anekdoti** [comment]
Hello @tg123 ,

thank you for your PR https://github.com/kubernetes-client/csharp/pull/1351. 

It solves the specific case that made me open issue https://github.com/kubernetes-client/csharp/issues/1350 where we observed an error with a timestamp with 9 nanosecond digits. 

However, as also described in my issue, timestamps can have **an arbitrary number (up to 9) of nanosecond digits** by the Golang implementation. For this reason, your PR only solves certain special cases whereas the PR on hand by @ndallmeyer solves the general case as by Kubernetes API conventions. 

To avoid further and foreseeable problems with deserializing timestamps, I would therefore strongly suggest to follow the general approach by @ndallmeyer .

**Comment by tg123** [comment]
@anekdoti could you please add bad test cases?
also please make pipeline happy
